pkg/scm/downloaders/file/download.go

```diff
@@ -54,9 +55,19 @@ func (f *File) Download(config *api.Config) (*git.SourceInfo, error) {
 		return nil, lerr
 	}
 
+	exclude, err := regexp.Compile(config.ExcludeRegExp)
```

Comment:
My first thought is that if `config.ExlcludeRegExp` is empty, the compiled regular expression will match EVERYTHING, so everything will get excluded?
I think you should check it's length first to make sure it's not empty before you compile/use it. And below on line 65 you will probably have to split that into separate checks, one for the `filesToIgnore[path]` and one for the regexp.

Reply:
Ah, yes, thanks. Empty string may be indeed problem.
Event with the bug locally most of the integration test passed.
And all unittest passed.

Reply:
The failure looked more like some flake.